data_collection/gazette/spiders/ro_jiparana.py

```diff
@@ -0,0 +1,420 @@
+    r"\s+de\s+"
+    rf"[^\d\s{punctuation}]+(?# nominal month in pt)"
+    r"\s+de\s+"
+    r"\d{4})(?# year)"
+    r"\s*[-–]+\s+"
+    r"Edição[^0-9]+"
+    r"([0-9]+)(?# edition_number)"
+    r"\s*[-–]?\s*"
+    rf"(\w?.*)(?# non-empty in case is_extra_edition_based_on_its_text)",
+    flags=re.IGNORECASE,
+)
+POST_2013_SPLIT_GAZETTE_TEXT_PATTERN = re.compile(
+    r"Edição[^0-9]+"
+    r"([0-9]+)(?# edition_number)"
+    r"\s*[-–]?\s*"
+    r"([^0-9]*)(?# non-empty in case is_extra_edition_based_on_its_text)"
+    r"\s+[--]\s+"
+    r"([0-9]{4})(?# gazette year)"
+    r"."
+    r"([0-9]{2})(?# gazette month)"
+    r"."
+    r"([0-9]{2})(?# gazette day)",
+    flags=re.IGNORECASE,
+)
+IS_EXTRA_EDITION_PATTERN = re.compile(
+    # sumplemento is a typo found in 2018-05-17
+    # suplmento is a typo found in 2018-06-20
+    # extraordinária found in 2020-09-05
+    r"(sum?ple?mento|extraordinária)",
+    flags=re.IGNORECASE,
+)
+DATE_2013_05_31 = date(
+    2013, 5, 31
+)  # Last day in 2013 before the split, which replaced the data rendering logic and format
+DISPLACED_DUPLICATE_EDITION_2013_03_08 = date(
+    2013, 3, 8
+)  # edition_number 1528 extra_edition from 2013-03-08 is duplicated in the post 2013 split page
+
+
+# The parameters the with GazettesDataPerPage inline comment rely on
+# a particular nested dict structure to temporarily store all the collected metadata.
+#
+# That storage is a dict in which the key is a tuple composed by 3 items:
+#   - gazette_date
+#   - edition_number
+#   - is_extra_edition
+#
+# As there can be multiple files for a single combination, the value of the outer dict is
+# another dict this time using the is_extra_edition_text as key and a set of urls
+# as the innermost value.
+#
+# Here are some examples where there are multiple files for the same day
+#
+# {
+#     (date(2010, 4, 15), '813', False): {
+#         'Parte 1': {
+#             'https://www.domjp.com.br/pdf/2010-04-15-part.1.pdf',
+#         },
+#         'Parte 2': {
+#             'https://www.domjp.com.br/pdf/2010-04-15-part.2.pdf',
+#         },
+#     },
+#
+#     (date(2010, 7, 14), '874', False): {
+#         '': {
+#             'https://www.domjp.com.br/pdf/2010-07-14.pdf',
+#         },
+#     },
+#     (date(2010, 7, 14), '874', True): {
+#         'Parte 2': {
+#             'https://www.domjp.com.br/pdf/2010-07-14-suplemento.pdf',
+#         },
+#     },
+#
+#     (date(2023, 1, 10), '3930', False): {
+#         '': {
+#             'https://diariooficialjp.com.br/pdf/2023-01-10.pdf',
+#         },
+#     },
+#     (date(2023, 1, 10), '3930', True): {
+#         'SUPLEMENTO': {
+#             'https://diariooficialjp.com.br/pdf/2023-01-10-suplemento.pdf',
+#         },
+#     },
+# }
+
+
+class RoJiParana(BaseGazetteSpider):
+    # Due to Python naming rules, we don't use the hyphen in Ji-Parana in the file name or class name
+    # However, scrapy allows having ji-parana as part of the spider's name
+    name = "ro_ji-parana"
```

Comment:
Aqui podemos manter com `_` mesmo. Outras ocorrências de hífen em nomes de municípios seriam substituídas por whitespace ou underline.
```suggestion
    name = "ro_ji_parana"
```

Reply:
Endereçado em a7780cd871f342f43fb0492b59d34ff889c68647